File: ios/ExpoAppExitModule.swift
import ExpoModulesCore

public class ExpoAppExitModule: Module {
    public func definition() -> ModuleDefinition {
        Name("ExpoAppExit")

        Function("AppExit") {
            DispatchQueue.main.asyncAfter(deadline: .now()) {
                UIApplication.shared.perform(#selector(NSXPCConnection.suspend))
                DispatchQueue.main.asyncAfter(deadline: .now() + 0.5) {
                    exit(0)
                }
            }
        }
    }
}

File: android/src/main/java/expo/modules/appexit/ExpoAppExitModule.kt
package expo.modules.appexit

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoAppExitModule : Module() {
    override fun definition() = ModuleDefinition {
        Name("ExpoAppExit")

        Function("AppExit") {
            val a = appContext.currentActivity
            a?.finishAffinity()
            Runtime.getRuntime().exit(0);
        }
    }
}
